vulcan/lib/client/jsx/components/workflow/session/primary_inputs.tsx

```diff
@@ -76,6 +77,10 @@ function PrimaryInputsInner() {
 
     if (Object.keys(withDefaults).length > 0) {
       setInputs((inputs) => ({...inputs, ...withDefaults}));
+      // Also check / trigger auto-committing after defaults are set
+      if (state.workflow && state.workflow.vignette && state.workflow.vignette.includes("Primary inputs are skippable")) {
```

Comment:
Trigger for primary inputs = this text anywhere in the vignette.

Reply:
Hm....that seems brittle? Is there a way to calculate this from the graph? I don't recall what the input step dependencies look like, but I imagine there should be a way to tell the UI that the primary inputs aren't needed until some future step...?

Reply:
In fact, if the server can provide some sort of dependency graph, could you calculate for all inputs (not just primary inputs) that a given UI step is skippable, without the CWL author having to explicitly name it as such?

Reply:
Certainly open to suggestions about these triggers, but I'm not sure I agree that a vulcan-side calculation of suitability is a good idea.  The issue with automating is that there's external context needed.  Primary defaults might appear valid (pass validations) yet be, say, tuned for only 1 of a few different use cases.  It's the workflow author who would have that context, so I think it's good to put the control into the workflow author's hands.  That's especially important in the primary input case because there's no user-toggle -- right as the user jumps in to their fresh session, if the check here passes, the primary inputs get committed, and if that works then steps are started running.

Difference between stepUI and primary inputs use cases:
- stepUI: "<target text>" at start of step 'doc' => show auto-commit toggle =user-marks-yes=> mark step to `triggerCommit` any future time that it gets marked as a bufferedStep (=validation-passes=> commited & marked to `triggerRun`)
- primary inputs: "<target text>" in workflow vignette => mark step as `triggerCommit` as soon as it's values are filled in (=validation-passes=> commited & marked to `triggerRun`)

Reply:
Hm, okay, I guess I'm not really familiar with the use cases, so very possible I misunderstood the intent of the PR! 

Would this have any effect on modifying a "skippable" step directly? Like every time you type into a text box, it commits and runs the workflow automatically instead of waiting for you to finish typing, if that step is skippable?

Reply:
I normally give a better description when I make a PR!  I'll add above.

>Would this have any effect on modifying a "skippable" step directly? Like every time you type into a text box, it commits and runs the workflow automatically instead of waiting for you to finish typing, if that step is skippable?

Shouldn't, no!  The triggers are essentially tied to "on-mount" of given step.  For primary inputs: at the start of the session, right after defaults are filled in.  For stepUIs: they're marked as a bufferedStep if they're being newly added to the set of shown input.  Those are the only times this system would be triggered.

Reply:
Ooooh no, I got some of that logic wrong... a committed step can become buffered if their inputs get changed.  I'll need to rework that trigger.